services/src/main/java/org/keycloak/utils/CRLUtils.java

```diff
@@ -51,15 +51,15 @@ public final class CRLUtils {
      * @throws GeneralSecurityException if some error in validation happens. Typically certificate not valid, or CRL signature not valid
      */
     public static void check(X509Certificate[] certs, X509CRL crl, KeycloakSession session) throws GeneralSecurityException {
-        if (certs.length < 2) {
-            throw new GeneralSecurityException("Not possible to verify signature on CRL. X509 certificate doesn't have CA chain available on it");
+        if (certs == null || certs.length < 1) {
+            throw new GeneralSecurityException("Not possible to verify signature on CRL because no certificate chain passed.");
         }
 
         X500Principal crlIssuerPrincipal = crl.getIssuerX500Principal();
         X509Certificate crlSignatureCertificate = null;
 
         // Try to find the certificate in the CA chain, which was used to sign the CRL
-        for (int i=1 ; i<certs.length ; i++) {
+        for (int i=0 ; i<certs.length ; i++) {
             X509Certificate currentCACert = certs[i];
```

Comment:
If the index starts at `0`, and the value is never used this might as well be a regular for loop.

```suggestion
        for (X509Certificate currentCACert: certs) {
```